src/platform/intel/ace/include/ace/lib/clk.h

```diff
@@ -20,30 +20,16 @@
 
 #include <ace/version.h>
 #include <sof/lib/cpu.h>
-#include <sof/lib/io.h>
-#include <sof/lib/memory.h>
-#include <stdint.h>
 
 struct sof;
 
-/** \brief Core(s) settings, up to CONFIG_CORE_COUNT */
-#define CLK_CPU(x)	(x)
-
-/** \brief SSP clock r-t settings are after the core(s) settings */
-#define CLK_SSP		CONFIG_CORE_COUNT
-
-/* SSP clock run-time data is the last one, so total number is ssp idx +1 */
-
 /** \brief Total number of clocks */
-#define NUM_CLOCKS	(CLK_SSP + 1)
+#define NUM_CLOCKS	NUM_CPU_FREQ
```

Comment:
this is a bit confusing. It's used in https://github.com/thesofproject/sof/blob/5389b257519c1fa00873386d94b70825dee04271/zephyr/lib/clk.c#L16 as a size of array, but then lines https://github.com/thesofproject/sof/blob/5389b257519c1fa00873386d94b70825dee04271/zephyr/lib/clk.c#L29-L32 iterate over that array, but using `CONFIG_CORE_COUNT` as the iteration number?

Reply:
Thanks @lyakh , this was a good catch. This is mixing number of clocks and number of frequencies (each CPU clock has). Fixed version uploaded now.